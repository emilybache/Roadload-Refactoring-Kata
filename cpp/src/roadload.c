
unsigned int calculateEnergyWithLoads(const struct CO2Values *co2Values) {
    return 0;
}
