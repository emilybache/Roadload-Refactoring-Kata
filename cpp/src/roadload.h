#ifndef ROADLOAD_H
#define ROADLOAD_H

#include <stdbool.h>

struct CO2Value {
    double value;
    bool hasValue;
};

struct CO2Values {
    struct CO2Value* High_roadload_f0;
    struct CO2Value* High_roadload_f1;
    struct CO2Value* High_roadload_f2;
    struct CO2Value* High_roadload_tm;
    struct CO2Value* High_roadload_combined;

    struct CO2Value* Mid_roadload_f0;
    struct CO2Value* Mid_roadload_f1;
    struct CO2Value* Mid_roadload_f2;
    struct CO2Value* Mid_roadload_tm;
    struct CO2Value* Mid_roadload_combined;

    struct CO2Value* Low_roadload_f0;
    struct CO2Value* Low_roadload_f1;
    struct CO2Value* Low_roadload_f2;
    struct CO2Value* Low_roadload_tm;
    struct CO2Value* Low_roadload_combined;

    bool IsMidPoint;
    bool IsSinglePoint;
};

void calculateEnergyWithLoads(struct CO2Values *co2Values);


#endif // ROADLOAD_H