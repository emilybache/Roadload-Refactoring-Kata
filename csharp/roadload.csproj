<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="packages\ApprovalTests.5.4.3\build\ApprovalTests.props" Condition="Exists('packages\ApprovalTests.5.4.3\build\ApprovalTests.props')" />
  <Import Project="packages\xunit.core.2.4.1\build\xunit.core.props" Condition="Exists('packages\xunit.core.2.4.1\build\xunit.core.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectGuid>{B92D4245-86CA-4380-BAEC-5560B749C31F}</ProjectGuid>
    <OutputType>Library</OutputType>
    <AppDesignerFolder>Properties</AppDesignerFolder>
    <RootNamespace>roadload</RootNamespace>
    <AssemblyName>Roadload</AssemblyName>
    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
    <FileAlignment>512</FileAlignment>
    <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">10.0</VisualStudioVersion>
    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
    <ReferencePath>$(ProgramFiles)\Common Files\microsoft shared\VSTT\$(VisualStudioVersion)\UITestExtensionPackages</ReferencePath>
    <IsCodedUITest>False</IsCodedUITest>
    <TestProjectType>UnitTest</TestProjectType>
    <TargetFrameworkProfile />
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug\</OutputPath>
    <DefineConstants>DEBUG;TRACE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
    <DebugType>pdbonly</DebugType>
    <Optimize>true</Optimize>
    <OutputPath>bin\Release\</OutputPath>
    <DefineConstants>TRACE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
  </PropertyGroup>
  <ItemGroup>
    <Reference Include="ApprovalTests, Version=3.0.0.0, Culture=neutral, PublicKeyToken=11bd7d124fc62e0f">
      <HintPath>packages\ApprovalTests.5.4.3\lib\net461\ApprovalTests.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="ApprovalUtilities, Version=3.0.0.0, Culture=neutral, PublicKeyToken=11bd7d124fc62e0f">
      <HintPath>packages\ApprovalUtilities.5.4.3\lib\net461\ApprovalUtilities.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="DiffEngine, Version=1.0.0.0, Culture=neutral, PublicKeyToken=c7a34512ecd69090">
      <HintPath>packages\DiffEngine.6.1.1\lib\net461\DiffEngine.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="EmptyFiles, Version=1.0.0.0, Culture=neutral, PublicKeyToken=c7a34512ecd69090">
      <HintPath>packages\EmptyFiles.2.0.5\lib\net461\EmptyFiles.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51">
      <HintPath>packages\Microsoft.Bcl.AsyncInterfaces.1.1.1\lib\net461\Microsoft.Bcl.AsyncInterfaces.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="Microsoft.Extensions.DependencyInjection.Abstractions, Version=3.1.4.0, Culture=neutral, PublicKeyToken=adb9793829ddae60">
      <HintPath>packages\Microsoft.Extensions.DependencyInjection.Abstractions.3.1.4\lib\netstandard2.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="Microsoft.Win32.Registry, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
      <HintPath>packages\Microsoft.Win32.Registry.4.7.0\lib\net461\Microsoft.Win32.Registry.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="mscorlib" />
    <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
      <HintPath>packages\Newtonsoft.Json.12.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
    </Reference>
    <Reference Include="System" />
    <Reference Include="System.Data" />
    <Reference Include="System.Data.SqlClient, Version=4.6.1.2, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
      <HintPath>packages\System.Data.SqlClient.4.8.2\lib\net461\System.Data.SqlClient.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="System.Drawing" />
    <Reference Include="System.Drawing.Common, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
      <HintPath>packages\System.Drawing.Common.4.5.0\lib\net461\System.Drawing.Common.dll</HintPath>
    </Reference>
    <Reference Include="System.Management" />
    <Reference Include="System.Runtime.CompilerServices.Unsafe, Version=4.0.4.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
      <HintPath>packages\System.Runtime.CompilerServices.Unsafe.4.5.3\lib\net461\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="System.Runtime.InteropServices.RuntimeInformation, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
      <HintPath>packages\System.Runtime.InteropServices.RuntimeInformation.4.3.0\lib\net45\System.Runtime.InteropServices.RuntimeInformation.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="System.Security.AccessControl, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
      <HintPath>packages\System.Security.AccessControl.4.7.0\lib\net461\System.Security.AccessControl.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="System.Security.Principal.Windows, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
      <HintPath>packages\System.Security.Principal.Windows.4.7.0\lib\net461\System.Security.Principal.Windows.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="System.Threading.Tasks.Extensions, Version=4.2.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51">
      <HintPath>packages\System.Threading.Tasks.Extensions.4.5.4\lib\net461\System.Threading.Tasks.Extensions.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="System.Xml" />
    <Reference Include="TextCopy, Version=1.0.0.0, Culture=neutral, PublicKeyToken=1ca091877d12ca03">
      <HintPath>packages\TextCopy.4.2.0\lib\net461\TextCopy.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c">
      <HintPath>packages\xunit.abstractions.2.0.3\lib\net35\xunit.abstractions.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c">
      <HintPath>packages\xunit.extensibility.core.2.4.1\lib\net452\xunit.core.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="xunit.execution.desktop, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c">
      <HintPath>packages\xunit.extensibility.execution.2.4.1\lib\net452\xunit.execution.desktop.dll</HintPath>
      <Private>True</Private>
    </Reference>
  </ItemGroup>
  <Choose>
    <When Condition="('$(VisualStudioVersion)' == '10.0' or '$(VisualStudioVersion)' == '') and '$(TargetFrameworkVersion)' == 'v3.5'">
      <ItemGroup>
        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />
      </ItemGroup>
    </When>
    <Otherwise>
      <ItemGroup>
        <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework" />
      </ItemGroup>
    </Otherwise>
  </Choose>
  <ItemGroup>
    <Compile Include="CO2Values.cs" />
    <Compile Include="Roadload.cs" />
    <Compile Include="RoadloadTest.cs" />
    <Compile Include="Properties\AssemblyInfo.cs" />
  </ItemGroup>
  <ItemGroup>
    <None Include="packages.config" />
  </ItemGroup>
  <ItemGroup>
    <Content Include="RoadloadTest.CalculateCO2Values.approved.txt" />
    <Content Include="RoadloadTest.CalculateCO2Values.received.txt" />
  </ItemGroup>
  <Choose>
    <When Condition="'$(VisualStudioVersion)' == '10.0' And '$(IsCodedUITest)' == 'True'">
      <ItemGroup>
        <Reference Include="Microsoft.VisualStudio.QualityTools.CodedUITestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
          <Private>False</Private>
        </Reference>
        <Reference Include="Microsoft.VisualStudio.TestTools.UITest.Common, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
          <Private>False</Private>
        </Reference>
        <Reference Include="Microsoft.VisualStudio.TestTools.UITest.Extension, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
          <Private>False</Private>
        </Reference>
        <Reference Include="Microsoft.VisualStudio.TestTools.UITesting, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
          <Private>False</Private>
        </Reference>
      </ItemGroup>
    </When>
  </Choose>
  <Import Project="$(VSToolsPath)\TeamTest\Microsoft.TestTools.targets" Condition="Exists('$(VSToolsPath)\TeamTest\Microsoft.TestTools.targets')" />
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
    <PropertyGroup>
      <ErrorText>This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105.The missing file is {0}.</ErrorText>
    </PropertyGroup>
    <Error Condition="!Exists('packages\xunit.core.2.4.1\build\xunit.core.props')" Text="$([System.String]::Format('$(ErrorText)', 'packages\xunit.core.2.4.1\build\xunit.core.props'))" />
    <Error Condition="!Exists('packages\xunit.core.2.4.1\build\xunit.core.targets')" Text="$([System.String]::Format('$(ErrorText)', 'packages\xunit.core.2.4.1\build\xunit.core.targets'))" />
    <Error Condition="!Exists('packages\EmptyFiles.2.0.5\build\EmptyFiles.targets')" Text="$([System.String]::Format('$(ErrorText)', 'packages\EmptyFiles.2.0.5\build\EmptyFiles.targets'))" />
    <Error Condition="!Exists('packages\ApprovalTests.5.4.3\build\ApprovalTests.props')" Text="$([System.String]::Format('$(ErrorText)', 'packages\ApprovalTests.5.4.3\build\ApprovalTests.props'))" />
    <Error Condition="!Exists('packages\ApprovalTests.5.4.3\build\ApprovalTests.targets')" Text="$([System.String]::Format('$(ErrorText)', 'packages\ApprovalTests.5.4.3\build\ApprovalTests.targets'))" />
  </Target>
  <Import Project="packages\xunit.core.2.4.1\build\xunit.core.targets" Condition="Exists('packages\xunit.core.2.4.1\build\xunit.core.targets')" />
  <Import Project="packages\EmptyFiles.2.0.5\build\EmptyFiles.targets" Condition="Exists('packages\EmptyFiles.2.0.5\build\EmptyFiles.targets')" />
  <Import Project="packages\ApprovalTests.5.4.3\build\ApprovalTests.targets" Condition="Exists('packages\ApprovalTests.5.4.3\build\ApprovalTests.targets')" />
  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
       Other similar extension points exist, see Microsoft.Common.targets.
  <Target Name="BeforeBuild">
  </Target>
  <Target Name="AfterBuild">
  </Target>
  -->
</Project>