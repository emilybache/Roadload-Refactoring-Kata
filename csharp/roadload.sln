
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio Version 16
VisualStudioVersion = 16.0.29326.143
MinimumVisualStudioVersion = 10.0.40219.1
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Roadload", "Roadload\Roadload.csproj", "{2E3CF952-E0A2-40DC-B1A8-B80D95B9EF9B}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Roadload.Tests", "Roadload.Tests\Roadload.Tests.csproj", "{725FE4B1-68AF-4682-969E-6925E2AC9EBF}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{2E3CF952-E0A2-40DC-B1A8-B80D95B9EF9B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{2E3CF952-E0A2-40DC-B1A8-B80D95B9EF9B}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{2E3CF952-E0A2-40DC-B1A8-B80D95B9EF9B}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{2E3CF952-E0A2-40DC-B1A8-B80D95B9EF9B}.Release|Any CPU.Build.0 = Release|Any CPU
		{725FE4B1-68AF-4682-969E-6925E2AC9EBF}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{725FE4B1-68AF-4682-969E-6925E2AC9EBF}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{725FE4B1-68AF-4682-969E-6925E2AC9EBF}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{725FE4B1-68AF-4682-969E-6925E2AC9EBF}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(ExtensibilityGlobals) = postSolution
		SolutionGuid = {576CD06C-AF6E-4D99-B731-5F51823109D8}
	EndGlobalSection
EndGlobal
