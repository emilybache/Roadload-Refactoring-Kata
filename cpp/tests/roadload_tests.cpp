
#include <climits>

#include "catch2/catch.hpp"
#include "ApprovalTests.hpp"
#include <string>

#include "printers.h"

extern "C"
{
#include "roadload.h"
}

TEST_CASE ("calculateEnergyWithLoads") {

    // These are the sample test data values listed in the README file
    auto *co2Values = new CO2Values();
    co2Values->High_roadload_f0 = new CO2Value{.value=3.5, .hasValue=true};
    co2Values->High_roadload_f1 = new CO2Value{.value=10.6575, .hasValue=true};
    co2Values->High_roadload_f2 = new CO2Value{.value=6.7, .hasValue=true};
    co2Values->High_roadload_tm = new CO2Value{.value=0.003, .hasValue=true};
    co2Values->Mid_roadload_f0 = new CO2Value{.value=3.2, .hasValue=true};
    co2Values->Mid_roadload_f1 = new CO2Value{.value=9.75, .hasValue=true};
    co2Values->Mid_roadload_f2 = new CO2Value{.value=5.9, .hasValue=true};
    co2Values->Mid_roadload_tm = new CO2Value{.value=0.0025, .hasValue=true};
    co2Values->Low_roadload_f0 = new CO2Value{.value=3.1, .hasValue=true};
    co2Values->Low_roadload_f1 = new CO2Value{.value=8.45, .hasValue=true};
    co2Values->Low_roadload_f2 = new CO2Value{.value=5.3, .hasValue=true};
    co2Values->Low_roadload_tm = new CO2Value{.value=0.002, .hasValue=true};

    // These are the values we will check afterwards
    co2Values->Low_roadload_combined = new CO2Value{.value=0, .hasValue=false};
    co2Values->Mid_roadload_combined = new CO2Value{.value=0, .hasValue=false};
    co2Values->High_roadload_combined = new CO2Value{.value=0, .hasValue=false};

    calculateEnergyWithLoads(co2Values);

    ApprovalTests::Approvals::verify(co2Values);
}