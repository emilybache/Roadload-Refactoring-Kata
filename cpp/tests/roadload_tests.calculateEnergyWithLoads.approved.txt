Low roadload combined: 0.000422223
Mid roadload combined: missing
High roadload combined: 0.000798758

